src/model/src/utils.py

```diff
@@ -50,44 +58,70 @@ def init(self):
 
         self.infected = random.sample(
             list(self.susceptible), self.initial_infected)
+        self.asymptomatic = random.sample(
+            list(self.infected), self.initial_asymptomatic)
         for infected_person in self.infected:
             self.susceptible.remove(infected_person)
+        for asymptomatic_person in self.asymptomatic:
+            self.infected.remove(asymptomatic_person)
 
     # Determine which category person belongs to
     def person_type(self, person):
         if person in self.susceptible:
             return 'susceptible'
         if person in self.infected:
             return 'infected'
+        if person in self.asymptomatic:
+            return 'asymptomatic'
         return 'recovered'
-
+   
     # Simulate new infected people
     def get_new_infected(self, infected_contact):
         new_infected = [
-            person for person in infected_contact if random.random() <= self.beta]
+            person for person in infected_contact & self.susceptible if random.random() <= self.beta]
```

Comment:
> What exactly are you checking here? Are you sure this will work? I don't think this will tell you if a person is present in both the sets.

Yeah, it's an intersection operation right ?, When i checked it long back, it was working fine

Reply:
Oh right.